lib/govspeak/remove_advisory_service.rb

```diff
@@ -0,0 +1,58 @@
+module Govspeak
+  class RemoveAdvisoryService
+    attr_reader :object, :regex, :whodunnit
+
+    def initialize(object, regex, whodunnit = User.find_by(name: "GDS Inside Government Team"))
+      @object = object
+      @regex = regex
+      @whodunnit = whodunnit
+    end
+
+    def process!
```

Comment:
I can't figure out how this method could be working. I think `Regexp.last_match(1)` is referencing the `@` rather than the content after the `@`. Should it be `Regexp.last_match(2)` ?

I'm wondering if using the string methods `scan` and `gsub` is making it hard to feel confident about what you're working with, because you have to remember that each subexpression in a `()` in your regexp equates to an object in the array that scan generates. If you'd used the Regexp method `match` it outputs the matches as a hash that you can [reference by key](https://ruby-doc.org/3.3.6/Regexp.html#class-Regexp-label-Named+Captures). The downside would be it only matches once, rather than searching the entire body, so for content with more than one advisory you'd need to think about recursion for example.

TLDR: 
- If you've had this code working with `scan` and `gsub` then stick with it, but please can you add comprehensive tests to show that your regex will match the different types of advisory markup that exists (eg @ only at the start, @ at the start and the end etc), and replace it as expected. 

- If you're struggling to get this to work, I wrote it using `Regexp.match` just to understand the shape of the problem, and I think it works so I could share that with you to see what you think...but I don't want to de-rail if you're almost there! So just let me know.

Reply:
YES PLEASE! A solution with `.match` would be so much better! I think this was the issue I was trying to fix in my tests! <3